kubectl-minio/cmd/tenant-debug.go

```diff
@@ -0,0 +1,141 @@
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+package cmd
+
+import (
+	"archive/zip"
+	"context"
+	"errors"
+	"fmt"
+	"github.com/minio/kubectl-minio/cmd/helpers"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"os"
+
+	miniov2 "github.com/minio/operator/pkg/apis/minio.min.io/v2"
+	"github.com/spf13/cobra"
+)
+
+const (
+	debugDesc = `'debug' command saves pod logs from a MinIO tenant`
+)
+
+type debugCmd struct {
+	out    io.Writer
+	errOut io.Writer
+	ns     string
+}
+
+func newTenantDebugCmd(out io.Writer, errOut io.Writer) *cobra.Command {
+	c := &debugCmd{out: out, errOut: errOut}
+
+	cmd := &cobra.Command{
+		Use:   "debug",
+		Short: "Save pod logs, events, and status for a tenant",
+		Long:  infoDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := c.validate(args); err != nil {
+				return err
+			}
+			klog.Info("info tenant command started")
+			err := c.run(args)
+			if err != nil {
+				klog.Warning(err)
+				return err
+			}
+			return nil
+		},
+	}
+	cmd = helpers.DisableHelp(cmd)
+	f := cmd.Flags()
+	f.StringVarP(&c.ns, "namespace", "n", helpers.DefaultNamespace, "namespace scope for this request")
+	return cmd
+}
+
+func (d *debugCmd) validate(args []string) error {
+	if args == nil {
+		return errors.New("provide the name of the tenant, e.g. 'kubectl minio tenant info tenant1'")
+	}
+	if len(args) != 1 {
+		return errors.New("info command supports a single argument, e.g. 'kubectl minio tenant info tenant1'")
+	}
+	if args[0] == "" {
+		return errors.New("provide the name of the tenant, e.g. 'kubectl minio tenant info tenant1'")
+	}
+	return nil
+}
+
+// run initializes local config and installs MinIO Operator to Kubernetes cluster.
+func (d *debugCmd) run(args []string) error {
+	// Create operator client
+	ctx := context.Background()
+	oclient, err := helpers.GetKubeOperatorClient()
+	if err != nil {
+		return err
+	}
+	client, err := helpers.GetKubeClient("")
+	if err != nil {
+		return err
+	}
+
+	tenant, err := oclient.MinioV2().Tenants(d.ns).Get(context.Background(), args[0], metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	listOpts := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", miniov2.TenantLabel, tenant.Name),
+	}
+	podsSet := client.CoreV1().Pods(tenant.Namespace)
+	events := client.CoreV1().Events(tenant.Namespace)
+	pods, err := podsSet.List(ctx, listOpts)
+	if err != nil {
+		return err
+	}
+	w, err := os.Create("dump.zip")
```

Comment:
take the tenant name here and call it `<tenant>-report.zip`

Reply:
Make sure to also open the file in `O_SYNC` to make sure to avoid any partial values stored